### DISCUSSION TITLE
Is it possible to add support for iOS 13, 14, and Safari 13 on my own?

### DISCUSSION THREAD

**Discussion by Hedmor** [chakra-ui/chakra-ui#8460] [Q&A] [answered]
Hello everyone! I developed a UIKit for my project and overlooked the fact that it needs to support iOS 13, 14, and Safari 13. I tried using core js, which partially solved the problem but not completely.

Here's what I have:

**iOS 13:** Offset due to an error within ChakraUI (NextJS 12).
![image](https://github.com/chakra-ui/chakra-ui/assets/47321719/c07eed4f-d480-423c-b18a-1655a04f7d52)
![image](https://github.com/chakra-ui/chakra-ui/assets/47321719/dae04b2d-1b05-4e5e-a392-50e4a1de066a)
**iOS 14.4:** Flex-gap generally works fine, but there are style issues. For example, borders around buttons.
![image](https://github.com/chakra-ui/chakra-ui/assets/47321719/a77840d1-22cd-4792-a9e5-9453a9f98ea2)
**Safari 13:** Flex gap doesn't work, and there are multiple style issues.
![image](https://github.com/chakra-ui/chakra-ui/assets/47321719/85f00c81-a2f0-4fbc-838c-e0b91d81513a)

No issues have been found in higher versions yet.

- Does anyone have experience with supporting ChakraUI on Safari 13 and iOS 13-14? 
- Is it worth overriding the styles of components, for example modifying <Stack ...> to support gap through margin? 
- Maybe there's an open-source project demonstrating how to do this with this UIKit? 
- What about ChakraUI 3, is there a risk that iOS and Safari versions will be updated, requiring us to make fixes again? Is there a way to automatically polyfill styles in JSS that I'm not aware of?

**Comment by segunadebayo** [comment, marked as answer]
We do not support iOS 13, iOS 14, or Safari 13. Flex `gap` and several selectors those browsers miss are required.

Stay on a newer Safari, or polyfill `gap` yourself. v3 will not add those polyfills.
